### PR TITLE
processor(cdc): remove dependency on the cdc self defined context

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: ref #6356 

### What is changed and how it works?

do not use `cdcContext.Context` to pass `captureInfo` and `changefeedID`, to reduce the usage of `cdcContext.Context`, and simplify the signature.

#### Tests 

 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: pkg/context/context.go [hunk 1]
```diff
@@ -212,13 +212,3 @@ func NewContext4Test(baseCtx context.Context, withChangefeedVars bool) Context {
 func NewBackendContext4Test(withChangefeedVars bool) Context {
 	return NewContext4Test(context.Background(), withChangefeedVars)
 }
-
-// ZapFieldCapture returns a zap field containing capture address
-func ZapFieldCapture(ctx Context) zap.Field {
```
Comment (sdojjy) [inline pkg/context/context.go:217]: can you remove these functions in ctx.go too?

**Review by sdojjy — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**